### PR TITLE
[SPARK-15979][SQL] Rename various Parquet support classes.

### PR DESCRIPTION
**Pull request by rxin**
## What changes were proposed in this pull request?

This patch renames various Parquet support classes from CatalystAbc to ParquetAbc. This new naming makes more sense for two reasons:
1. These are not optimizer related (i.e. Catalyst) classes.
2. We are in the Spark code base, and as a result it'd be more clear to call out these are Parquet support classes, rather than some Spark classes.
## How was this patch tested?

Renamed test cases as well.

### CODE REVIEW

**Comment by rxin** [comment]
JIRA is down - I will create a ticket when it comes back up.

cc @liancheng

File: mllib/src/main/scala/org/apache/spark/ml/source/libsvm/LibSVMRelation.scala [hunk 1]
```diff
@@ -141,7 +141,7 @@ class LibSVMFileFormat extends TextBasedFileFormat with DataSourceRegister {
       sparkSession: SparkSession,
       options: Map[String, String],
       files: Seq[FileStatus]): Map[String, String] = {
-    def computeNumFeatures(): Int = {
+    val numFeatures = options.get("numFeatures").filter(_.toInt > 0).getOrElse {
```
Comment (rxin) [inline mllib/src/main/scala/org/apache/spark/ml/source/libsvm/LibSVMRelation.scala]: cc @cloud-fan / @liancheng 

I got rid of the inline function.

**Comment by cloud-fan** [comment]
LGTM

**Comment by liancheng** [comment]
LGTM pending Jenkins.

**Comment by rxin** [comment]
Merging in master/2.0.

**Comment by rxin** [comment]
I merged this into master but there is a conflict in 2.0. Going to submit a separate pr.

**Comment by rxin** [comment]
See https://github.com/apache/spark/pull/13700

**Pull request closed without merging**